### PR TITLE
Promote NU3043 warning to error in dotnet nuget sign command

### LINKED ISSUE #13814: Deprecate SHA-1 usage in .NET 10
Related - https://github.com/NuGet/docs.microsoft.com-nuget/pull/3339

This DCR has to be merged into dev after 6.12 branch has been created.

### PR DESCRIPTION
**Pull request by kartheekp-ms**
# Bug

Fixes: https://github.com/NuGet/Home/issues/13814

## Description

Updated dotnet nuget sign command to accept fingerprints from the SHA-2 family (SHA256, SHA384, or SHA512) instead of SHA-1. If a SHA-1 fingerprint or invalid value is passed, the commands raise an [NU3043](https://learn.microsoft.com/nuget/reference/errors-and-warnings/nu3043) error indicating that SHA-1 is insecure. 

## PR Checklist

- [x] Meaningful title, helpful description and a linked NuGet/Home issue
- [x] Added tests
- [x] Link to an issue or pull request to update docs if this PR changes settings, environment variables, new feature, etc. - https://github.com/NuGet/Home/issues/13963

### CODE REVIEW

**Review by jeffkl — DISMISSED** [review]
You'll need to fix the compiler error

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Strings.resx [hunk 1]
```diff
@@ -971,7 +971,8 @@ Non-HTTPS access will be removed in a future version. Consider migrating to 'HTT
     <value>Allows HTTP connections for adding or updating packages. Note: This method is not secure. For secure options, see https://aka.ms/nuget-https-everywhere for more information.</value>
   </data>
   <data name="SignCommandInvalidCertificateFingerprint" xml:space="preserve">
-    <value>Invalid value for '--certificate-fingerprint' option. The value must be a SHA-256, SHA-384, or SHA-512 certificate fingerprint (in hexadecimal).</value>
+    <value>{0}: Invalid value for '--certificate-fingerprint' option. The value must be a SHA-256, SHA-384, or SHA-512 certificate fingerprint (in hexadecimal).</value>
```
Comment (jgonz120) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/Strings.resx:974]: looks like you have a test failing regarding this message

Comment (kartheekp-ms) [reply src/NuGet.Core/NuGet.CommandLine.XPlat/Strings.resx:974]: Yeah, I have fixed the tests failing due to string mismatch.

**Review by jeffkl — APPROVED** [review]

**Comment by aortiz-msft** [comment]
@kartheekp-ms - This is a .NET SDK breaking change, so we need to follow the corresponding process.

**Pull request merged**